syntax = "proto3";

message ScrapeRequest {
  repeated string urls = 1;
}

message ScrapeResponse {
  message ScrapedUrl {
    string base_url = 1;
    string rtsp_url = 2;
  }
  repeated ScrapedUrl scraped_urls = 1;
}

service Scraper {
  rpc Scrape(ScrapeRequest) returns (ScrapeResponse);
}